pennylane/templates/subroutines/qsvt.py

```diff
@@ -0,0 +1,356 @@
+    the :math:`Wx` or rotation convention.
+
+    Depending on the choice of convention for blockencoding, the same phase angles will produce different
+    polynomial transformations. We provide the functionality to swap between blockencoding conventions and
+    to transform the phase angles accordingly using the :code:`convention` keyword argument.
+
+    .. seealso::
+
+        :class:`~.QSVT` and `A Grand Unification of Quantum Algorithms <https://arxiv.org/pdf/2105.02859.pdf>`_.
+
+    Args:
+        A (tensor_like): the general :math:`(n \times m)` matrix to be encoded
+        angles (tensor_like): a list of angles by which to shift to obtain the desired polynomial
+        wires (Iterable[int, str], Wires): the wires the template acts on
+        convention (string): can be set to ``"Wx"`` to convert quantum signal processing angles in the
+            `Wx` convention to QSVT angles.
+
+    **Example**
+
+    To implement QSVT in a circuit, we can use the following method:
+
+    >>> dev = qml.device("default.qubit", wires=2)
+    >>> A = np.array([[0.1, 0.2], [0.3, 0.4]])
+    >>> angles = np.array([0.1, 0.2, 0.3])
+    >>> @qml.qnode(dev)
+    ... def example_circuit(A):
+    ...     qml.qsvt(A, angles, wires=[0, 1])
+    ...     return qml.expval(qml.PauliZ(wires=0))
+
+    The resulting circuit implements QSVT.
+
+    >>> print(qml.draw(example_circuit)(A))
+    0: ─╭QSVT─┤  <Z>
+    1: ─╰QSVT─┤
+
+    To see the implementation details, we can expand the circuit:
+
+    >>> q_script = qml.tape.QuantumScript(ops=[qml.qsvt(A, angles, wires=[0, 1])])
+    >>> print(q_script.expand().draw(decimals=2))
+    0: ─╭∏_ϕ(0.10)─╭BlockEncode(M0)─╭∏_ϕ(0.20)─╭BlockEncode(M0)†─╭∏_ϕ(0.30)─┤
+    1: ─╰∏_ϕ(0.10)─╰BlockEncode(M0)─╰∏_ϕ(0.20)─╰BlockEncode(M0)†─╰∏_ϕ(0.30)─┤
+    """
+    if qml.math.shape(A) == () or qml.math.shape(A) == (1,):
+        A = qml.math.reshape(A, [1, 1])
+
+    c, r = qml.math.shape(A)
+
+    UA = BlockEncode(A, wires=wires, do_queue=False)
+    projectors = []
+
+    if convention == "Wx":
+        angles = _qsp_to_qsvt(angles)
+        global_phase = (len(angles) - 1) % 4
+
+        if global_phase:
+            qml.exp(qml.Identity(wires=wires), 0.5j * np.pi * (4 - global_phase))
+
+    for idx, phi in enumerate(angles):
+        dim = c if idx % 2 else r
+        projectors.append(PCPhase(phi, dim=dim, wires=wires, do_queue=False))
+
+    projectors = projectors[::-1]  # reverse order to match equation
+    return QSVT(UA, projectors, wires=wires)
+
+
+class QSVT(Operation):
+    r"""QSVT(UA,projectors,wires)
+    Implements the
+    `quantum singular value transformation <https://arxiv.org/abs/1806.01838>`__ (QSVT) circuit.
+
+    Given an :class:`~.Operator` :math:`U`, which block encodes the matrix :math:`A`, and a list of
+    projector-controlled phase shift operations :math:`\vec{\Pi}_\phi`, this template applies a
+    circuit for the quantum singular value transformation as follows.
+
+    .. note ::
```

Comment:
Note should be earlier. Here it's weird that it says "as follows" then we read the note

Reply:
Moved the note up here and in lower-case `qsvt` for consistency.